src/RoslynPad.Common.UI/ViewModels/DocumentViewModel.cs

```diff
@@ -160,17 +185,80 @@ private ObservableCollection<DocumentViewModel> ReadChildren()
         {
             return new ObservableCollection<DocumentViewModel>(
                 IOUtilities.EnumerateDirectories(Path)
-                .Select(x => new DocumentViewModel(x, isFolder: true))
-                .OrderBy(OrderByName)
-                    .Concat(IOUtilities.EnumerateFiles(Path, "*" + DefaultFileExtension)
-                        .Select(x => new DocumentViewModel(x, isFolder: false))
-                        .Where(x => !x.IsAutoSave)
-                        .OrderBy(OrderByName)));
+                    .Concat(IOUtilities.EnumerateFiles(Path, "*" + DefaultFileExtension))
+                    .Select(x => new DocumentViewModel(x, _documentFileWatcher))
+                    .Where(x => !x.IsAutoSave)
+                    .OrderBy(dd => !dd.IsFolder)
+                    .ThenBy(OrderByName));
         }
 
         private static string OrderByName(DocumentViewModel x)
         {
             return Regex.Replace(x.Name, "[0-9]+", m => m.Value.PadLeft(100, '0'));
         }
+
+        private static Func<IEnumerable<DocumentViewModel>, IOrderedEnumerable<DocumentViewModel>> SortPredicate =>
```

Comment:
Encapsulate adding a child into a private `AddChild` method, and inline this lambda.